server/job/blacklistedTagsProcessor.ts

```diff
@@ -0,0 +1,186 @@
+import type { SortOptions } from '@server/api/themoviedb';
+import { SortOptionsIterable } from '@server/api/themoviedb';
+import type {
+  TmdbSearchMovieResponse,
+  TmdbSearchTvResponse,
+} from '@server/api/themoviedb/interfaces';
+import { MediaType } from '@server/constants/media';
+import dataSource from '@server/datasource';
+import { Blacklist } from '@server/entity/Blacklist';
+import Media from '@server/entity/Media';
+import type {
+  RunnableScanner,
+  StatusBase,
+} from '@server/lib/scanners/baseScanner';
+import { getSettings } from '@server/lib/settings';
+import logger from '@server/logger';
+import { createTmdbWithRegionLanguage } from '@server/routes/discover';
+import type { EntityManager } from 'typeorm';
+
+const TMDB_API_DELAY_MS = 250;
+class AbortTransaction extends Error {}
+
+class BlacklistedTagProcessor implements RunnableScanner<StatusBase> {
+  private running = false;
+  private progress = 0;
+  private total = 0;
+
+  public async run() {
+    this.running = true;
+
+    try {
+      await dataSource.transaction(async (em) => {
+        await this.cleanBlacklist(em);
```

Comment:
Why do you remove all previous blacklisted items before creating new entries?
The job to create blacklist entries takes a bit of time, so it means that in the meantime no tag will be blacklisted anymore?

Reply:
Since it's in a transaction, there isn't a time where things are unexpectedly missing from the blacklist, both the clean and the new entries are executed at the same time when the transaction commits.

I remove all entries blacklisted by tag because the job's goal is to keep the first X pages of every search order for each tag in the blacklist. Particularly for new releases, I felt it was important to keep entries up to date without continuing the dump things into the blacklist.